client/data/followers/remove-follower.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useMutation, useQueryClient } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+
+function useRemoveFollower() {
+	const queryClient = useQueryClient();
+	const { mutate: removeFollower, ...rest } = useMutation(
+		( { siteId, follower, type } ) => {
+			const method = type === 'email' ? 'removeEmailFollower' : 'removeFollower';
+			return wp.undocumented().site( siteId )[ method ]( follower.ID );
+		},
+		{
+			onSuccess( data, variables ) {
```

Comment:
Do we also need to handle the follower removal failure case somehow?

Reply:
Yes we should totally handle that case. Although not at this place. Error handling should be done where the custom hook is used depending on the returned `error` object.

```js
// function component body
const { removeFollower, error } = useRemoveFollower();

React.useEffect( () => {
  if ( error ) {
    doSomething();
  }
}, [ error ])
```

As far as I understand these `onSuccess`, `onError` are mainly used to handle hook internal state or data (like optimistic updates, etc).

I'm still thinking about a good solution to error handling in this specific case (there are a few gotchas which are specific to Calypso). Would you mind if I follow up with another PR? Error handling for removing followers seems to be lacking on prod as well so that'd be a great improvement.

Reply:
Makes sense, I don't mind with following up in another PR at all 👍